ironfish/src/wallet/account.ts

```diff
@@ -195,6 +195,89 @@ export class Account {
     })
   }
 
+  async addPendingTransaction(
+    transaction: Transaction,
+    decryptedNotes: Array<DecryptedNote>,
+    submittedSequence: number | null,
+    tx?: IDatabaseTransaction,
+  ): Promise<void> {
+    await this.walletDb.db.withTransaction(tx, async (tx) => {
+      let balanceDelta = 0n
+      for (const decryptedNote of decryptedNotes) {
+        if (decryptedNote.forSpender) {
+          continue
+        }
+
+        await this.walletDb.nonChainNoteHashes.put([this.prefix, decryptedNote.hash], null, tx)
+
+        const note = new Note(decryptedNote.serializedNote)
+        await this.walletDb.saveDecryptedNote(
+          this,
+          decryptedNote.hash,
+          {
+            accountId: this.id,
+            note,
+            spent: false,
+            transactionHash: transaction.hash(),
+            nullifier: null,
+            index: null,
+            blockHash: null,
+            sequence: null,
+          },
+          tx,
+        )
+
+        balanceDelta += note.value()
+      }
+
+      for (const spend of transaction.spends()) {
```

Comment:
If you have a chain of TX like `A -> B` and get B first, this won't mark any notes from A as spent. This code doesn't check notes and check if nullifiers exist in a pending state here. We should add a test for this, and fix this type of case in the future.